### PR TITLE
ZCS-7512 : Click action on buttons present in mail opens the mail client in the reading pane, when zimbra_use_owasp_html_sanitizer=TRUE

### PR DESCRIPTION
**Pull request by auminpatel**

### CODE REVIEW

File: WebRoot/js/zimbraMail/mail/view/ZmMailMsgView.js [hunk 1]
```diff
@@ -987,6 +987,10 @@ function(params) {
 
 	var html = params.html || "";
 	
+	if(html.search(/(<form)(?![^>]+action)(.*?>)/g)){
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/mail/view/ZmMailMsgView.js]: tabbing is not proper

File: WebRoot/js/zimbraMail/mail/view/ZmMailMsgView.js [hunk 2]
```diff
@@ -987,6 +987,10 @@ function(params) {
...
+        html=html.replace(/(<form)(?![^>]+action)(.*?>)/g, `$1 action="SAMEHOSTFORMPOST-BLOCKED" target="_blank"$2`)
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/mail/view/ZmMailMsgView.js]: we should remove target attribute if already present
currently it does something like this `<form action="SAMEHOSTFORMPOST-BLOCKED" target="_blank" target="inline"></form>` if target is already present

File: WebRoot/js/zimbraMail/mail/view/ZmMailMsgView.js (hunk 1 above)
Comment (auminpatel) [reply WebRoot/js/zimbraMail/mail/view/ZmMailMsgView.js]: Done

File: WebRoot/js/zimbraMail/mail/view/ZmMailMsgView.js (hunk 2 above)
Comment (auminpatel) [reply WebRoot/js/zimbraMail/mail/view/ZmMailMsgView.js]: Resolved

File: WebRoot/js/zimbraMail/mail/view/ZmMailMsgView.js (hunk 1 above)
Comment (auminpatel) [reply WebRoot/js/zimbraMail/mail/view/ZmMailMsgView.js]: Done

File: WebRoot/js/zimbraMail/mail/view/ZmMailMsgView.js [hunk 3]
```diff
@@ -987,6 +987,18 @@ function(params) {
...
 	
+	if(html.search(/(<form)(?![^>]+action)(.*?>)/g)){
+		html = html.replace(/(<form)(?![^>]+action)(.*?>)/ig, function(form){
+				if(form.match(/target/g)){
+					form=form.replace(/(<.*)(target=.*)(.*>)/g,`$1action="SAMEHOSTFORMPOST-BLOCKED" target="_blank"$3`);
+				}
+				else{
+					form=form.replace(/(<form)(?![^>]+action)(.*?>)/g, `$1 action="SAMEHOSTFORMPOST-BLOCKED" target="_blank"$2`)
+				}
+		return form;
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/mail/view/ZmMailMsgView.js]: tabbing not proper

Comment (auminpatel) [reply WebRoot/js/zimbraMail/mail/view/ZmMailMsgView.js]: Done

**Pull request merged**